src/fsharp/service/service.fsi

```diff
@@ -470,6 +497,31 @@ type public FSharpChecker =
     /// <param name="userOpName">An optional string used for tracing compiler operations associated with this request.</param>
     member CheckFileInProject : parsed: FSharpParseFileResults * filename: string * fileversion: int * source: string * options: FSharpProjectOptions * ?textSnapshotInfo: obj * ?userOpName: string -> Async<FSharpCheckFileAnswer>
 
+    /// <summary>
+    /// <para>
+    ///   Check a source code file, returning a handle to the results
+    /// </para>
+    /// <para>
+    ///    Note: all files except the one being checked are read from the FileSystem API
+    /// </para>
+    /// <para>
+    ///   Return FSharpCheckFileAnswer.Aborted if a parse tree was not available.
+    /// </para>
+    /// </summary>
+    ///
+    /// <param name="parsed">The results of ParseFile for this file.</param>
+    /// <param name="filename">The name of the file in the project whose source is being checked.</param>
+    /// <param name="fileversion">An integer that can be used to indicate the version of the file. This will be returned by TryGetRecentCheckResultsForFile when looking up the file.</param>
+    /// <param name="source">The full source for the file.</param>
+    /// <param name="options">The options for the project or script.</param>
+    /// <param name="textSnapshotInfo">
+    ///     An item passed back to 'hasTextChangedSinceLastTypecheck' (from some calls made on 'FSharpCheckFileResults') to help determine if 
+    ///     an approximate intellisense resolution is inaccurate because a range of text has changed. This 
+    ///     can be used to marginally increase accuracy of intellisense results in some situations.
+    /// </param>
+    /// <param name="userOpName">An optional string used for tracing compiler operations associated with this request.</param>
+    member CheckFileInProject : parsed: FSharpParseFileResults * filename: string * fileversion: int * source: Source * options: FSharpProjectOptions * ?textSnapshotInfo: obj * ?userOpName: string -> Async<FSharpCheckFileAnswer>
```

Comment:
On looking at this I'm beginning to think we should just remove the `string` overloads and take `Source` everywhere, and all callers will have to adjust.

Reply:
Yes, I will remove the added overloads and just make the change to the API directly to take in `Source`